File: 4-MedianOfTwoSortedArrays/4-MedianOfTwoSortedArrays.py
# Last updated: 04/10/2025, 01:47:00
'''class Solution(object):
    def findMedianSortedArrays(self, nums1, nums2):
        k = []
        p = 0

        for i in nums1:
            k.append(i)
        for j in nums2:
            k.append(j)

        k.sort()
        for u in range(1, len(k) + 1):
            p = p + 1

        if p % 2 == 0:
            n = p // 2
            median = (k[n - 1] + k[n]) / 2
            return float('{:.5f}'.format(median))
        else:
            n = p // 2
            return float('{:.5f}'.format(k[n]))'''

''' Example usage:
solution = Solution()
nums1 = [1, 3]
nums2 = [2]
print(solution.findMedianSortedArrays(nums1, nums2))  # Output: 2.0
'''
class Solution(object):
    def findMedianSortedArrays(self, nums1, nums2):
        k = []
        p = 0

        for i in nums1:
            k.append(i)
        for j in nums2:
            k.append(j)

        k.sort()
        for u in range(1, len(k) + 1):
            p = p + 1

        if p % 2 == 0:
            n = p // 2
            median = (k[n - 1] + k[n]) / 2.0
            return median
        else:
            n = p // 2
            return float(k[n])

File: 4-MedianOfTwoSortedArrays/4-MedianOfTwoSortedArrays.java
// Last updated: 04/10/2025, 01:46:54
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
       ArrayList<Integer> s = new ArrayList<>();
        for (int i = 0; i < nums1.length; i++) {
            s.add(nums1[i]);
        }
        for (int i = 0; i < nums2.length; i++) {
            s.add(nums2[i]);
        }
        Collections.sort(s);
        double ans;
        int m = s.size() / 2;
        if (s.size() % 2 == 0) {
            ans = (double) (s.get(m - 1) + s.get(m)) / 2;
        } else {
            ans = (double) s.get(m);
        }
        return ans;

    }
}